swap/src/protocol/alice/cancel.rs

```diff
@@ -0,0 +1,76 @@
+use crate::bitcoin::{ExpiredTimelocks, Txid, Wallet};
+use crate::database::{Database, Swap};
+use crate::protocol::alice::AliceState;
+use anyhow::{bail, Result};
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, thiserror::Error, Clone, Copy)]
+pub enum Error {
+    #[error("The cancel timelock has not expired yet because the timelock has not expired. Please try again later")]
+    CancelTimelockNotExpiredYet,
+}
+
+pub async fn cancel(
+    swap_id: Uuid,
+    bitcoin_wallet: Arc<Wallet>,
+    db: Arc<Database>,
+    force: bool,
+) -> Result<Result<(Txid, AliceState), Error>> {
+    let state = db.get_state(swap_id)?.try_into_alice()?.into();
+
+    let (monero_wallet_restore_blockheight, transfer_proof, state3) = match state {
+
+        // In case no XMR has been locked, move to Safely Aborted
+        AliceState::Started { .. }
+        | AliceState::BtcLocked { .. } => bail!("Cannot cancel swap {} because it is in state {} where no XMR was locked. Try aborting instead.", swap_id, state),
```

Comment:
The idea was that I hint to use the `safely-abort` command here. But I don't want to literally put the command name here, because it might become outdated. 

I removed the hint `Try aborting instead.` now, because I think it is not really important and no matter what it might become outdated. 
A power user should be aware of this option anyway...